src/crimson/os/seastore/cache.h

```diff
@@ -641,10 +646,32 @@ class Cache {
     }
   };
 
+  struct trans_byextent_efforts_t {
+    std::array<effort_t, EXTENT_TYPES_MAX> read_by_ext;
+    std::array<effort_t, EXTENT_TYPES_MAX> mutate_by_ext;
+    std::array<uint64_t, EXTENT_TYPES_MAX> delta_bytes_by_ext;
+    std::array<effort_t, EXTENT_TYPES_MAX> retire_by_ext;
+    std::array<effort_t, EXTENT_TYPES_MAX> fresh_by_ext;
+
+    auto& get_by_name(const std::string& effort_name) {
+      if (effort_name == "READ") {
+        return read_by_ext;
+      } else if (effort_name == "MUTATE") {
+        return mutate_by_ext;
+      } else if (effort_name == "RETIRE") {
+        return retire_by_ext;
+      } else {
+        ceph_assert(effort_name == "FRESH");
```

Comment:
i'd suggest do not use `ceph_assert()` to detect programming/logic errors. use `assert()` in this case.

Reply:
fixed